crates/glue/src/load.rs

```diff
@@ -130,46 +130,19 @@ pub fn generate(
                     }
 
                     let lib = unsafe { Library::new(lib_path) }.unwrap();
-                    type MakeGlueReturnType =
-                        roc_std::RocResult<roc_std::RocList<roc_type::File>, roc_std::RocStr>;
-                    type MakeGlue = unsafe extern "C" fn(
-                        *mut RocCallResult<MakeGlueReturnType>,
-                        &roc_std::RocList<roc_type::Types>,
-                    );
-
-                    let make_glue: libloading::Symbol<MakeGlue> = unsafe {
-                        lib.get("roc__makeGlueForHost_1_exposed_generic".as_bytes())
-                            .unwrap_or_else(|_| panic!("Unable to load glue function"))
-                    };
                     let roc_types: roc_std::RocList<roc_type::Types> =
                         types.iter().map(|x| x.into()).collect();
-                    let mut files =
-                        RocCallResult::new(roc_std::RocResult::err(roc_std::RocStr::empty()));
-                    unsafe { make_glue(&mut files, &roc_types) };
-
-                    // Roc will free data passed into it. So forget that data.
-                    std::mem::forget(roc_types);
-
-                    let files: Result<roc_std::RocList<roc_type::File>, roc_std::RocStr> =
-                        match Result::from(files) {
-                            Err((msg, tag)) => match tag {
-                                CrashTag::Roc => panic!(r#"Roc failed with message: "{msg}""#),
-                                CrashTag::User => panic!(r#"User crash with message: "{msg}""#),
-                            },
-                            Ok(x) => x.into(),
-                        };
-
-                    let files = files.unwrap_or_else(|err| {
-                        eprintln!("Glue generation failed: {err}");
-
-                        process::exit(1);
-                    });
+
+                    // NOTE: DO NOT DROP LIB! the return value will include static roc strings that
```

Comment:
Haha. wow. That's a good catch.

Reply:
yeah, I was not careful and that will just segfault. even a later explicit drop of the lib causes issues, so something to keep in mind.